### PR TITLE
Updated import wrapper and IO module; added specman to IO

### PR DESCRIPTION
**Pull request by thcasey3**
Updated importing and made example, unittests, and GUI compatible

### CODE REVIEW

**Comment by ylin00** [comment]
@thcasey3 Is the `data/topspin/hydrationGUI Results` generated by any script or needed as input?

File: data/topspin/hydrationGUI Results/E_ksig.csv
Comment (ylin00) [inline data/topspin/hydrationGUI Results/E_ksig.csv]: Could you change `/hydrationGUI Results/` to `hydrationGUI_Results/`, to avoid potential issues with space in the path.

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]: mising `else`

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]: need initializing `c`

Maybe `c = ""`

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]:
```suggestion
            params[c + "_" + str(i)] = splt_exp_content
        else:
            pass

```

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]: Do you need close and reopen here?

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]: `else: y_data = ?`

File: dnplab/dnpIO/topspin.py
Comment (ylin00) [inline dnplab/dnpIO/topspin.py]:
```suggestion
def load_title(path="1", titlePath="pdata/1", titleFilename="title"):
```

File: dnplab/dnpIO/topspin.py
Comment (ylin00) [inline dnplab/dnpIO/topspin.py]:
```suggestion
def load_acqu(path="1", paramFilename="acqus"):
```

File: dnplab/dnpIO/topspin.py
Comment (ylin00) [inline dnplab/dnpIO/topspin.py]:
```suggestion
def load_proc(path="1", procNum=1, paramFilename="procs"):
```

File: dnplab/dnpIO/topspin.py
Comment (ylin00) [inline dnplab/dnpIO/topspin.py]: do you need `os.path.join` ?

File: dnplab/dnpIO/topspin.py
Comment (ylin00) [inline dnplab/dnpIO/topspin.py]: Delete the lines instead of comment it out.

File: unittests/test_dnpImport.py
Comment (ylin00) [inline unittests/test_dnpImport.py]: 👍

File: unittests/test_dnpImport.py
Comment (ylin00) [inline unittests/test_dnpImport.py]:
```suggestion
            wrapper.load(os.path.join(self.test_data, "%i" % expNum, "data.csv"), data_type="prospa")
```

**Review by ylin00 — CHANGES_REQUESTED** [review]

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]: This is potentially error-prone. In windows `os.sep == \` but `path[-1]` can still be `/` 
```suggestion
    path = os.path.normpath(path)
```
Need tests

File: data/topspin/hydrationGUI Results/E_ksig.csv
Comment (thcasey3) [reply data/topspin/hydrationGUI Results/E_ksig.csv]: Done

File: dnplab/dnpIO/specman.py
Comment (thcasey3) [reply dnplab/dnpIO/specman.py]: done

File: dnplab/dnpIO/specman.py
Comment (thcasey3) [reply dnplab/dnpIO/specman.py]: done

File: dnplab/dnpIO/specman.py
Comment (thcasey3) [reply dnplab/dnpIO/specman.py]: done

File: dnplab/dnpIO/specman.py
Comment (thcasey3) [reply dnplab/dnpIO/specman.py]: i'll limit it to 4D and just raise an error message if more than 4

File: dnplab/dnpIO/topspin.py
Comment (thcasey3) [reply dnplab/dnpIO/topspin.py]: I fixed this in many more places as well

File: dnplab/dnpIO/topspin.py
Comment (thcasey3) [reply dnplab/dnpIO/topspin.py]: done

File: unittests/test_dnpImport.py
Comment (thcasey3) [reply unittests/test_dnpImport.py]: done

File: dnplab/dnpIO/specman.py
Comment (thcasey3) [reply dnplab/dnpIO/specman.py]: I added normpath to the wrapper function so the path is normalized before it gets to any of the IO functions

File: dnplab/dnpIO/specman.py
Comment (thcasey3) [reply dnplab/dnpIO/specman.py]: If I don't reopen the file it appears that the file_opened variable is changed by np.fromfile, the second call to np.fromfile doesn't find what it needs from file_opened. The only way I can get it to work is to open it twice. Maybe there is a better way to do it but I haven't found it yet.

Comment (ylin00) [reply dnplab/dnpIO/specman.py]: I see. I think it is good enough.

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]: no need to return after raise Exception.
```suggestion
```

File: dnplab/dnpIO/specman.py
Comment (ylin00) [inline dnplab/dnpIO/specman.py]:
```suggestion
```

File: dnplab/hydrationGUI.py
Comment (ylin00) [inline dnplab/hydrationGUI.py]: I didn't review changes to this script. Please let me know if you have specific things that I can help to check.

**Review by ylin00 — APPROVED** [review]

File: dnplab/hydrationGUI.py
Comment (thcasey3) [reply dnplab/hydrationGUI.py]: Ok, I think everything is in order. All tests pass, the example runs, and the GUI works fine

**Review by ylin00 — APPROVED** [review]

**Comment by ylin00** [comment]
If possible, please limit the number of updates to the pull request. Each time it will cost ~10min of the 2000min/month Github Actions quota.

**Comment by ylin00** [comment]
@thcasey3 Please consider creating a new pull request instead of adding new major edits to an existing one.

File: dnplab/dnpNMR.py
Comment (ylin00) [inline dnplab/dnpNMR.py]:
```suggestion
        raise TypeError("type must be either 'poly' or 'expon'")
```

File: dnplab/dnpNMR.py
Comment (ylin00) [inline dnplab/dnpNMR.py:330]:
```suggestion
```

File: dnplab/dnpNMR.py
Comment (ylin00) [inline dnplab/dnpNMR.py]: use `warnings.warn` instead of print.

**Review by ylin00 — APPROVED** [review]

File: dnplab/dnpNMR.py
Comment (thcasey3) [reply dnplab/dnpNMR.py]: done

File: dnplab/dnpNMR.py
Comment (thcasey3) [reply dnplab/dnpNMR.py:330]: done

File: dnplab/dnpNMR.py
Comment (thcasey3) [reply dnplab/dnpNMR.py]: done

**Review by ylin00 — CHANGES_REQUESTED** [review]
This PR contains multiple edits. Have to be processed after approval of #49 .

**Review by ylin00 — APPROVED** [review]

**Review by tkeller12 — APPROVED** [review]

**Pull request merged**
